bloomfilter/Data/BloomFilter/Hash.hs

```diff
@@ -24,125 +25,50 @@ module Data.BloomFilter.Hash
-
-foreign import ccall unsafe "lookup3.h _jenkins_hashwordOff" hashWordBA
-    :: ByteArray# -> CSize -> CSize -> Word32 -> IO Word32
-
-foreign import ccall unsafe "lookup3.h _jenkins_hashword2" hashWord2
-    :: Ptr Word32 -> CSize -> Ptr Word32 -> Ptr Word32 -> IO ()
-
-foreign import ccall unsafe "lookup3.h _jenkins_hashlittle" hashLittle
-    :: Ptr a -> CSize -> Word32 -> IO Word32
-
-foreign import ccall unsafe "lookup3.h _jenkins_hashlittleOff" hashLittleBA
-    :: ByteArray# -> CSize -> CSize -> Word32 -> IO Word32
-
-foreign import ccall unsafe "lookup3.h _jenkins_hashlittle2" hashLittle2
-    :: Ptr a -> CSize -> Ptr Word32 -> Ptr Word32 -> IO ()
+foreign import capi unsafe "HsXXHash.h hs_XXH3_64bits_withSeed_offset"
+    unsafe_xxh3_withSeed_ba :: ByteArray# -> CSize -> CSize -> Word64 -> Word64
 
 class Hashable a where
-    -- | Compute a 32-bit hash of a value.  The salt value perturbs
-    -- the result.
-    hashIO32 :: a               -- ^ value to hash
-             -> Word32          -- ^ salt
-             -> IO Word32
-
     -- | Compute a 64-bit hash of a value.  The first salt value
     -- perturbs the first element of the result, and the second salt
     -- perturbs the second.
-    hashIO64 :: a               -- ^ value to hash
-             -> Word64           -- ^ salt
-             -> IO Word64
-    hashIO64 v salt = do
-                   let s1 = fromIntegral (salt `unsafeShiftR` 32) .&. maxBound
-                       s2 = fromIntegral salt
-                   h1 <- hashIO32 v s1
-                   h2 <- hashIO32 v s2
-                   return $ (fromIntegral h1 `unsafeShiftL` 32) .|. fromIntegral h2
-
--- | Compute a 32-bit hash.
-hash32 :: Hashable a => a -> Word32
-hash32 = hashSalt32 0x16fc397c
+    hashSalt64
+        :: Word64               -- ^ salt
+        -> a           -- ^ value to hash
+        -> Word64
 
+-- | Compute a 64-bit hash.
 hash64 :: Hashable a => a -> Word64
-hash64 = hashSalt64 0x16fc397cf62f64d3
-
--- | Compute a salted 32-bit hash.
-hashSalt32 :: Hashable a => Word32  -- ^ salt
-           -> a                 -- ^ value to hash
-           -> Word32
-{-# INLINE hashSalt32 #-}
-hashSalt32 salt k = unsafePerformIO $ hashIO32 k salt
-
--- | Compute a salted 64-bit hash.
-hashSalt64 :: Hashable a => Word64  -- ^ salt
-           -> a                 -- ^ value to hash
-           -> Word64
-{-# INLINE hashSalt64 #-}
-hashSalt64 salt k = unsafePerformIO $ hashIO64 k salt
-
--- | Compute a list of 32-bit hashes.  The value to hash may be
--- inspected as many times as there are hashes requested.
-hashes :: Hashable a => Int     -- ^ number of hashes to compute
-       -> a                     -- ^ value to hash
-       -> [Hash]
-hashes n v = unfoldr go (n,0x3f56da2d)
-    where go (k,s) | k <= 0    = Nothing
-                   | otherwise = let s' = hashSalt32 s v
-                                 in Just (s', (k-1,s'))
+hash64 = hashSalt64 0
```

Comment:
because any salt is good as any other.

Also `XXH3_64bits(ptr, len) = SSH3_64bits_withSeed(ptr, len, 0)`, see https://xxhash.com/doc/v0.8.2/group___x_x_h3__family.html#ga0c06e71549705effd58cb7bd0969307e